pkg/restore/restore.go

```diff
@@ -1813,7 +1826,26 @@ func (ctx *restoreContext) getSelectedRestoreableItems(resource, targetNamespace
 			continue
 		}
 
-		if !ctx.selector.Matches(labels.Set(obj.GetLabels())) {
+		var skipItem = false
+		var skip = 0
+		ctx.log.Infof("selectors specified: %s for item: %s", ctx.selectors, item)
+		for _, s := range ctx.selectors {
+			if !s.Matches(labels.Set(obj.GetLabels())) {
+				skip++
+			}
+
+			if len(ctx.selectors) == skip && skip > 0 {
+				ctx.log.Infof("setting skip flag to true for item: %s", item)
+				skipItem = true
+			}
+		}
+
+		if skipItem {
+			ctx.log.Infof("restore selector labels did not match, skipping restore of item: %s", skipItem, item)
```

Comment:
In this case, the `selector` is ignored.  Is it the design that when en(ctx.selectors) > 0 the ctx.selector is ignored?

Reply:
Yes, we would want to process the `orLabelSelectors` first and based on that decide whether to skip the restore of the item or not. It would not make sense if the item was to be skipped based on `orLabelSelectors` and was to be restored based on `LabelSelector`. This will also preserve the existing velero behavior where the `orLabelSelectors` do not exist.